sys/stdvga.c

```diff
@@ -88,6 +91,45 @@ static int stdvga_palette_write(vga_device_t *vga, uio_t *uio) {
   return 0;
 }
 
+static int stdvga_get_resolution(vga_device_t *vga, uint16_t *xres,
+                                 uint16_t *yres, unsigned int *bpp) {
+  stdvga_device_t *stdvga = STDVGA_FROM_VGA(vga);
+  *xres = stdvga->width;
+  *yres = stdvga->height;
+  *bpp = stdvga->bpp;
+  return 0;
+}
+
+static int stdvga_set_resolution(vga_device_t *vga, uint16_t xres,
+                                 uint16_t yres, unsigned int bpp) {
+  stdvga_device_t *stdvga = STDVGA_FROM_VGA(vga);
+
+  /* I have not tested larger resolutions yet. */
+  if (xres > 640 || yres > 480)
```

Comment:
You could either browse an array of predefined video modes or autodetect them somehow...